Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -283,12 +285,12 @@ Ref<TimeRanges> MediaSource::seekable()
 
     // ↳ If duration equals NaN:
     // Return an empty TimeRanges object.
-    if (m_duration.isInvalid())
+    if (isClosed() || m_duration.isInvalid())
```

Comment:
This is not in the spec.
If this is to handle the case of stopped sources, maybe we can set m_duration to invalid in stop() method.

Reply:
actually, it is kind in the spec, `If the [readyState](https://w3c.github.io/media-source/#dom-mediasource-readystate) attribute is "[closed](https://w3c.github.io/media-source/#dom-readystate-closed)" then return NaN and abort these steps.` so if closed duration is NaN

Reply:
I've reworked MediaSource::duration so that it gets queried via a method; this will make it duration() returns invalid if the media source is closed (which is per spec)